CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs

```diff
@@ -319,6 +327,135 @@ private void ExportListOfSolutionsToBeMerged(OrganizationServiceProxy servicePro
             }
         }
 
+        /// <summary>
+        /// Method gets deployment instance record
+        /// </summary>
+        /// <param name="serviceProxy">organization service proxy</param>
+        /// <param name="solutionFile">solution file info</param>        
+        private void GetDeploymentInstance(OrganizationServiceProxy serviceProxy, SolutionFileInfo solutionFile)
+        {
+            Entity sourceControl = solutionFile.Solution;
+            EntityCollection deploymentInstance = FetchDeplopymentInstance(serviceProxy, sourceControl.Id);
+
+            if (deploymentInstance.Entities.Count > 0)
+            {
+                foreach (Entity instance in deploymentInstance.Entities)
```

Comment:
you can do threading here, use parallel.foreach

Reply:
In future. we will start using parallel.foreach.